### DISCUSSION TITLE
Export original sample format

### DISCUSSION THREAD

**Discussion by ferro4prez** [B0ney/xmodits#22] [Ideas]
Hi! This is unreal - I'm wondering if rather than exporting to WAV if it's possible to export to the original file format?  This would be super useful in generating a usable samples folder from a bunch of MOD files, to load directly into ProTracker for example.

**Comment by B0ney** [comment]
Hi,

From your provided example, by "original file format", I'm assuming you're referring to the Amiga IFF (8SVX) format? Sure, I can add that.

I'm currently rewriting the core library to make adding features like this easier, so it'll take a *little* longer.

**Comment by ferro4prez** [comment]
Man that would be so great!

Imagining a world where I feed it my entire MOD library, and it generates a
folder of usable 8SVX files that can be loaded directly into ProTracker!

**Comment by ferro4prez** [comment]
Hey just got this when ripping a big folder, and thought I'd share :

[image: image.png]

Just an idea, is it possible for a corrupt module to simply be
skipped/aborted when it's detected, rather than taking down the whole
program?  There's no indication of what file crashed it, and where to
resume from etc.

Thank you for your amazing work with this, you can't begin to imagine how
useful this is!

**Comment by B0ney** [comment]
> Is it possible for a corrupt module to simply be
skipped/aborted when it's detected, rather than taking down the whole
program

Yes, that's what it should normally do. This is a bug. 
For some reason, the image isn't showing up. Can you file an [issue](https://github.com/B0ney/xmodits/issues/new/choose) and upload it there?

> There's no indication of what file crashed it, and where to
resume from etc.

I plan to add this for the next version, you're free to file another issue with the **enhancement** tag as a reminder.

Quick question, were the majority of the modules Amiga MOD files? That format was a bit weird to implement because they're very little ways to validate it, so I may have missed something. 

I know I may be asking a lot from you, but could you upload an archive of the folder you tried to rip? 
The folder might be large, but 7zip can compress the folder and split them up to manageable chunks (25 - 50MB).

You can upload them to a cloud service like mega, mediafire or even include them in your bug report if it's not that big.

**If you can't (or don't feel comfortable to) do this, don't worry about it.** I can do more testing with the modules from modarchive.

Having tested this program with ~8000  modules, I'm quite interested.

**Comment by ferro4prez** [comment]
Hey! Thanks for the response - I'll comment on GitHub to keep it all
tracked and part of the process.

And yeah funny, I downloaded the entire modarchive torrent (aboug 50gigs
lol), and wrote PowerShell to sort it into folders A-Z... I believe I ran
it on the "A" folder in this case, which probably had tens of thousands of
MODs.  I can still try to recreate the issue and share the folder though!

One last question - is there any standard in terms of sample rate?  I
noticed the samples that were extracted from MODs specifically all seemed
to be pitched an octave down when played at the default sample rate.


...maybe I'll share what I"m up to.  I'm trying to extract every single
sample from every module in the modarchive, and load them into XLN XO... XO
- XLN Audio <https://www.xlnaudio.com/products/xo> ... generating a "cloud"
of Amiga samples all intelligently clustered by XO based on if they're a
kick, snare, perc, hihat, sfx etc.   If the samples are pitched
incorrectly, a snare will probably be misinterpreted as a kick, etc.

Really appreciate your efforts in keeping these incredible formats alive!

**Reply by B0ney** [reply]
> One last question - is there any standard in terms of sample rate?  I
noticed the samples that were extracted from MODs specifically all seemed
to be pitched an octave down when played at the default sample rate.

The sample rate for ripped MODs is ``8363 Hz``. I intended to calculate the sample rate from its [finetune](https://github.com/a740g/QB64-MOD-Player/blob/main/docs/MOD.txt#L37), but I got lazy and just left it as a placeholder.
Do note the ripping process doesn't do any destructive resampling, it's just metadata put into the WAV.

I'm personally not happy with how I've implemented the MOD format. So it's on my to-do list.

The core rewrite is going pretty smoothly. I don't have as much free time now, but even at such an early stage it's so much nicer to work with. The rewrite has multithreading from the beginning, so extracting samples from larger modules should be a bit faster.

Looking forward to your updates!

**Comment by ferro4prez** [comment]
Amazing! Legit man despite you not being happy with the implementation,
you've by far made the most functional app out there for this purpose.

And thanks for the details, that finetune tweak would be butter!

Talk soon, I'll hit GitHub rn.

**Comment by B0ney** [comment]
Hi,

I know it's been a couple of months, but I'm almost finished developing version ``0.10.0``.

Here's the pre-release: https://github.com/B0ney/xmodits/releases/tag/v0.10.0-rc1

Along with a multitude of features, this new version should **produce better sounding samples for MOD files**.

The reason they originally sounded low pitched was because they were at a lower octave, *not* because of its finetune. This has been fixed :)

If you have the time to test it, I'd greatly appreciate any feedback before I publish it.

**Comment by ferro4prez** [comment]
Thank you so much, trying it now!

[image: image.png]

....lolllll!  I wonder how long it'll take ProTracker to load 122,455
folders.

**Comment by ferro4prez** [comment]
Once this is done, gonna load it into XLN Audio's "XO" where it'll build a
cloud map of all the samples based on sonic profile, for use in modern DAWs.

This is a dream come true man, thank you for your efforts.

[image: image.png]

Training an AI is next hahah.  We live in good times.
